sdk/identity/identity/src/credentials/visualStudioCodeCredential.ts

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { TokenCredential, AccessToken, GetTokenOptions } from "@azure/core-auth";
+
+import fs from "fs";
+import os from "os";
+import path from "path";
+
+import { CredentialUnavailableError } from "../client/errors";
+import { IdentityClient, TokenCredentialOptions } from "../client/identityClient";
+import { AzureAuthorityHosts } from "../constants";
+import { checkTenantId } from "../util/checkTenantId";
+import { credentialLogger, formatError, formatSuccess } from "../util/logging";
+import { VSCodeCredentialFinder } from "./visualStudioCodeCredentialExtension";
+
+const CommonTenantId = "common";
+const AzureAccountClientId = "aebc6443-996d-45c2-90f0-388ff96faa56"; // VSC: 'aebc6443-996d-45c2-90f0-388ff96faa56'
```

Comment:
where does this magic come from anyway?

Reply:
This client ID is consistent for other languages: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/identity/Azure.Identity/src/VisualStudioCodeCredential.cs#L23 it’s like the Azure CLI client ID, it’s a special app for public flows that is designed to be used to authenticate with VSCode.

Reply:
Like Daniel said, this is a public AppID that can authenticate a user principal in any tenant so long as the tenant has public flows enabled.